projects/core/src/product/facade/product-search-by-code.service.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { select, Store } from '@ngrx/store';
+import { auditTime, map, Observable, tap, using } from 'rxjs';
+import { Product } from '../../model';
+import { ProductActions } from '../store/actions/index';
+import { StateWithProduct } from '../store/product-state';
+import { ProductSelectors } from '../store/selectors/index';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ProductSearchByCodeService {
+  constructor(protected store: Store<StateWithProduct>) {}
```

Comment:
major: For DI, let's use the `inject` function to prevent breaking changes in the constructor signature if a new dependency is required. See: https://wiki.one.int.sap/wiki/display/spar/Coding+Guidelines#CodingGuidelines-Useinject()functioninsteadofconstructorforDependencyInjection

Reply:
Done